src/librustc_trans/debuginfo/gdb.rs

```diff
@@ -22,19 +23,13 @@ use std::ptr;
 use syntax::attr;
 
 
-/// Inserts a side-effect free instruction sequence that makes sure that the
-/// .debug_gdb_scripts global is referenced, so it isn't removed by the linker.
-pub fn insert_reference_to_gdb_debug_scripts_section_global(ccx: &CrateContext, builder: &Builder) {
+/// Inserts the .debug_gdb_scripts global into the set of symbols
+/// to be placed in `llvm.used`, so it isn't removed by the linker.
```

Comment:
Actually, I have proven myself wrong. If I use the extremely obscure output section type of `INFO` in the linker script, then *exactly* the thing that I expect happens!
```
.debug_gdb_scripts 0 (INFO) : {
  KEEP(*(.debug_gdb_scripts))
}
```
```
  1 .debug_gdb_scripts 000000aa  00000000  00000000  00011848  2**0
                  CONTENTS, READONLY, DEBUGGING
```
I do not believe it is viable to integrate this into upstream rustc and so this PR and its parent issue may be closed.